bazel/BUILD

```diff
@@ -873,3 +873,8 @@ platform(
         "@build_bazel_apple_support//constraints:device",
     ],
 )
+
+cc_library(
+    name = "python_headers",
```

Comment:
trying to understand how this will work

i think the suggestion i made before - ie `"@system_python//:python_headers"` points to locally installed headers 

i wasnt 100% about the suggestion - i saw it in the rules_python source code and various places, and the following ~works

```console
$ bazel build @system_python//:python_headers
```

altho one concern i had was that the system python in the build image 3.10 whereas we are now using 3.11

fwiw there is some work ongoing in `rules_python` to add hermetic py headers to the toolchain

Reply:
iiuc this wont be used/tested in normal circumstances and im not sure its needed at all any more

Reply:
I'm not 100% sure what target is referencing the `:python_headers` as a dependency. `:python_headers` need to actually provide real header files if there is some C++ code that needs python.h which we do not have. I saw that other projects in similar boat just create this dummy target to satisfy bazel. protobuf was doing this this before for us, but now they have removed it so I just defined it locally.

Reply:
ok - checking the (old) upstream and this is basically what it did also (altho it had some notes about what was needed to make it work)

Reply:
I've added comment. Hopefully we will never need to make this target supply real headers.